### PR TITLE
major: Refactor and split out

### PR DESCRIPTION
**Pull request by klcodanr**
Changes:

 - Reduced scope of ingestor client to only support submitting ingestion requests
 - Added a new BatchExecutor to traverse trees and/or batches and asynchronously process applicable items
 - Cleaned up some minor test issues
 - Removed CLI and related artifacts
 - Moved rest, secret and context capabilities to `@adobe/content-lake-commons`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@adobe/content-lake-extractors-shared",
   "version": "1.9.0",
-  "description": "Content Lake - Shared",
+  "description": "Content Lake - Extractors Shared",
```
Comment (jdelbick) [inline package.json:4]: maybe remove `Rename package from @adobe/content-lake-extractors-shared to @adobe/content-lake-shared` from the description :)

File: src/ingestor.js [hunk 2]
```diff
@@ -10,44 +10,67 @@
  * governing permissions and limitations under the License.
  */
 
-import { fetch as originalFetch } from '@adobe/fetch';
-import { forEachLimit, mapLimit } from 'async';
-import fetchBuilder from 'fetch-retry';
+import { FetchRetry } from '@adobe/content-lake-commons';
```
Comment (mfrisbey) [inline src/ingestor.js:13]: I haven't looked at the commons bundle yet - did we modify `fetch-retry` in that package?

**Review by mfrisbey — APPROVED** [review]
Looks great! I like the class-based approach, and the new BatchProcess that seems to handle the actual process of extracting and ingesting assets. I left a few comments and suggestions, but nothing I feel like would warrant stopping a merge.

File: src/ingestor.js (hunk 2 above)
Comment (klcodanr) [reply src/ingestor.js:13]: Yep, essentially the retry logic was pulled out of here into commons.

File: src/batch-provider.js
[the remark this answers is not in the document]
Comment (klcodanr) [reply src/batch-provider.js:1]: I can separate it out, but I also wanted a "NoOp" version of this for testing as well as if you only needed to override certain methods.

**Pull request merged**